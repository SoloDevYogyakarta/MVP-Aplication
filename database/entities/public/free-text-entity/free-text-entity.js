"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.freeTextEntity = exports.FreeTextEntity = void 0;
const sequelize_1 = require("sequelize");
const base_entity_1 = require("../../base-entity/base-entity");
const entity_1 = require("../../entity");
class FreeTextEntity extends base_entity_1.BaseEntity {
}
exports.FreeTextEntity = FreeTextEntity;
exports.freeTextEntity = entity_1.sequelize.define('TEXT', {
    text: sequelize_1.DataTypes.TEXT,
    sparepart_id: sequelize_1.DataTypes.INTEGER,
}, {
    tableName: 'TEXT',
    schema: 'public',
});
//# sourceMappingURL=free-text-entity.js.map