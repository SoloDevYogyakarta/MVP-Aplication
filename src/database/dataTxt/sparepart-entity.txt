{"id":3,"public_id":"AHA1qU8mgu_mHSx1KmZhr","text":"Audeo comminor comptus. Similique demo aufero convoco caries demens. Degero ipsam ulterius copia tamquam decipio adipiscor talis pecto."}