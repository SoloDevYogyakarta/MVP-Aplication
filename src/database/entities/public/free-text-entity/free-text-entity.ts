import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
import { sequelize } from '../../entity';

export class FreeTextEntity extends BaseEntity {
  text: string;
  sparepart_id: number;
}

export type FreeTextCreationAttribute = Optional<FreeTextEntity, 'id'>;
export interface FreeTextInstance
  extends Model<FreeTextCreationAttribute, FreeTextEntity>,
    FreeTextEntity {}
export const freeTextEntity = sequelize.define<FreeTextInstance>(
  'TEXT',
  {
    text: DataTypes.TEXT,
    sparepart_id: DataTypes.INTEGER,
  } as ModelAttributes<FreeTextInstance>,
  {
    tableName: 'TEXT',
    schema: 'public',
  },
);
