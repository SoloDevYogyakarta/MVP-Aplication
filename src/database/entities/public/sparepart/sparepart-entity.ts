import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
import { sequelize } from '../../entity';

export class SparepartEntity extends BaseEntity {
  text: string;
}

export type SparepartCreationAttribute = Optional<SparepartEntity, 'id'>;
export interface SparepartInstance
  extends Model<SparepartCreationAttribute, SparepartEntity>,
    SparepartEntity {}
export const sparepartEntity = sequelize.define<SparepartInstance>(
  'SPAREPART',
  {
    public_id: DataTypes.STRING,
    text: DataTypes.STRING,
  } as ModelAttributes<SparepartInstance>,
  {
    tableName: 'SPAREPART',
    schema: 'public',
    timestamps: false,
  },
);
