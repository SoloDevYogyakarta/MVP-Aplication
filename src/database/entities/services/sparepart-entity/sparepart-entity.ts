import { Model, ModelAttributes, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
import { sequelize } from '../../entity';

export class SparepartEntity extends BaseEntity {}

export type SparePartCreationAttribute = Optional<SparepartEntity, 'id'>;
export interface SparepartInstance
  extends Model<SparePartCreationAttribute, SparepartEntity>,
    SparepartEntity {}
export const sparepartEntity = sequelize.define<SparepartInstance>(
  'SPAREPART',
  {} as ModelAttributes<SparepartInstance>,
  {
    tableName: 'SPAREPART',
    schema: 'SERVICES',
  },
);
