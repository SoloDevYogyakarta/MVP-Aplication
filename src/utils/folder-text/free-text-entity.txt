{"id":7,"text":"Benigne acies culpa. Incidunt currus corona derideo nesciunt talus asporto synagoga celer comes. Sint patrocinor aer complectus comis iusto avarus.","sparepart_id":7,"updatedAt":"2023-11-07T07:15:15.837Z","createdAt":"2023-11-07T07:15:15.837Z"}