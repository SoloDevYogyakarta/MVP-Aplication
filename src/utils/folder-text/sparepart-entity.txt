{"id":7,"updatedAt":"2023-11-07T07:15:15.814Z","createdAt":"2023-11-07T07:15:15.814Z"}