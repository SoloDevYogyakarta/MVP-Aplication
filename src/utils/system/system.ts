import fs from 'fs';
import { join } from 'path';

function joinpath(path: string): string {
  return join(__dirname, path);
}

function readpath(path: string): string {
  return fs.readFileSync(joinpath(path), { encoding: 'utf-8' });
}

function createpath(path: string, data: unknown): void {
  let values = data;
  if (typeof values !== 'string') {
    values = JSON.stringify(values);
  }
  fs.writeFileSync(joinpath(path), values as string);
}

function removepath(path: string): void {
  fs.unlinkSync(joinpath(path));
}

export { joinpath, readpath, createpath, removepath };
