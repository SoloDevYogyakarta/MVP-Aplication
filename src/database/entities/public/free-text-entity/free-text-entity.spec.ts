import { faker } from '@faker-js/faker';
import { ModelCtor } from 'sequelize';
import { createpath } from '../../../../utils/system/system';
import { sparepartEntity } from '../../services/sparepart-entity/sparepart-entity';
import { freeTextEntity, FreeTextInstance } from './free-text-entity';

describe('FreeTextEntity', () => {
  let entity: ModelCtor<FreeTextInstance>;

  beforeEach(() => {
    entity = freeTextEntity;
  });

  it('should to be defined', () => expect(entity).toBeDefined());

  it('render correctly', () => expect(entity).toMatchSnapshot());

  it('create', async () => {
    const sparepart = await sparepartEntity.create();
    sparepart.save();
    const create = await entity.create({
      text: faker.lorem.paragraph(),
      sparepart_id: sparepart.id,
    });
    create.save();
    createpath('../folder-text/sparepart-entity.txt', sparepart);
    createpath('../folder-text/free-text-entity.txt', create);
    expect(create.sparepart_id).toEqual(sparepart.id);
  });
});
