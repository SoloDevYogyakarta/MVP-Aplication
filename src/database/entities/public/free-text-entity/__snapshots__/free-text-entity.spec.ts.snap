// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FreeTextEntity render correctly 1`] = `[Function]`;
